File: Javascript/Array/E-225. Implement Stack using Queues.ts
class MyStack {
  firstQueue: number[];
  secondQueue: number[];

  constructor() {
    this.firstQueue = [];
    this.secondQueue = [];
  }

  moveItems = (sourceQueue: number[], targetQueue: number[]): void => {
    while (sourceQueue.length) {
      const item = sourceQueue.shift();
      if (item) targetQueue.push(item);
    }
  };

  push(x: number): void {
    this.moveItems(this.secondQueue, this.firstQueue);
    this.firstQueue.push(x);
  }

  pop(): number {
    while (this.firstQueue.length > 1) {
      const item = this.firstQueue.shift();
      if (item) this.secondQueue.push(item);
    }
    /** remaining item */
    const item = this.firstQueue.shift();
    this.moveItems(this.secondQueue, this.firstQueue);
    return item || 0;
  }

  top(): number {
    return this.firstQueue[this.firstQueue.length - 1];
  }

  empty(): boolean {
    return this.firstQueue.length === 0;
  }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */

var obj = new MyStack();
obj.push(1);
obj.push(2);
var param_2 = obj.pop();
var param_3 = obj.top();
var param_4 = obj.empty();

File: Python/Array/E-225. Implement Stack using Queues.py
class MyStack:

    def __init__(self):
        self.first_queue = []
        self.second_queue = []

    def move_items(self, source_queue: list[int], target_queue: list[int]):
        while len(source_queue):
            target_queue.append(source_queue.pop(0))

    def push(self, x: int) -> None:
        self.move_items(self.second_queue, self.first_queue)
        self.first_queue.append(x)

    def pop(self) -> int:
        while len(self.first_queue) > 1:
            self.second_queue.append(self.first_queue.pop(0))
        item = self.first_queue.pop(0)
        self.move_items(self.second_queue, self.first_queue)
        return item

    def top(self) -> int:
        return self.first_queue[len(self.first_queue) - 1]

    def empty(self) -> bool:
        return len(self.first_queue) == 0

        # Your MyStack object will be instantiated and called as such:
        # obj = MyStack()
        # obj.push(x)
        # param_2 = obj.pop()
        # param_3 = obj.top()
        # param_4 = obj.empty()
